.github/workflows/main.yml

```diff
@@ -27,30 +27,57 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        pair:
-          - erlang: 22.2
-            elixir: 1.10.1
-          - erlang: 22.1
-            elixir: 1.9.4
-          - erlang: 21.3
-            elixir: 1.8.2
-          - erlang: 20.3.1
-            elixir: 1.7.4
-          - erlang: 19.3
-            elixir: 1.6.6
-          - erlang: 18.3
+        include:
+          - otp-version: 18.3.4.11
+            elixir: 1.5.3
+          - otp-version: 20.3.8.26
             elixir: 1.5.3
+          - otp-version: 19.3.6.13
+            elixir: 1.6.6
+          - otp-version: 21.3.8.16
+            elixir: 1.6.6
+          - otp-version: 19.3.6.13
+            elixir: 1.7.4
+          - otp-version: 22.3.4.2
+            elixir: 1.7.4
+          - otp-version: 20.3.8.26
+            elixir: 1.8.2
+          - otp-version: 22.3.4.2
+            elixir: 1.8.2
+          - otp-version: 20.3.8.26
+            elixir: 1.9.4
+          - otp-version: 22.3.4.2
+            elixir: 1.9.4
+          - otp-version: 21.3.8.16
+            elixir: 1.10.3
+          - otp-version: 23.0.2
+            elixir: 1.10.3
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2.3.1
 
       - name: Install OTP and Elixir
-        uses: actions/setup-elixir@v1
+        uses: actions/setup-elixir@v1.3
         with:
-          otp-version: ${{matrix.pair.erlang}}
-          elixir-version: ${{matrix.pair.elixir}}
+          otp-version: ${{matrix.otp-version}}
+          elixir-version: ${{matrix.elixir}}
 
       - name: Install dependencies
-        run: mix deps.get
+        run: mix deps.get --only test
+
+      - name: Compile dependencies
+        run: mix deps.compile
+
+      - name: Compile application
+        if: matrix.elixir != '1.10.3'
+        run: mix compile
+        env:
+          MIX_ENV: test
+
+      - name: Compile application treating warnings as errors (only for last Elixir version)
+        if: matrix.elixir == '1.10.3'
+        run: mix compile --warnings-as-errors
+        env:
+          MIX_ENV: test
```

Comment:
Can't we make this as another task, like we do formatting? That is, explicitly add a task for compiling with warning on latest Erlang/Elixir.

Reply:
Sure can do, but why do you believe that would be better?

Reply:
Because I'm not a fan of the `if`ing around. I'd prefer "Compile with warnings" to be a different task. Thanks!

Reply:
A little less DRY, but the `if`s are gone, as requested: https://github.com/whatyouhide/stream_data/pull/142/commits/a5682fd1770a367bee96d4252f618bee5316279d.